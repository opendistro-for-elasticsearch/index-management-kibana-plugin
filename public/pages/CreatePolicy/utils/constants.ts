/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "hot",
      states: [
        {
          name: "hot",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "cold", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "cold",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
    },
  },
  null,
  4
);
