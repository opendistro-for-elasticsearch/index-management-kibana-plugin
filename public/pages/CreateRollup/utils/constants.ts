/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

//Added last_updated_time and start_time to avoid error, but they should be system generated.
//Added a delay as 0, but this should be optional, so null should be fine.
export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "2ms",
          timezone: "Africa/Abidjan",
        },
      },
    ],
    enabled: false,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];
