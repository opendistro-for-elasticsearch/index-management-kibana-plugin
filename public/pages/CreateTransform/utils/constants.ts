/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const EMPTY_TRANSFORM = JSON.stringify({
  transform: {
    description: "",
    enabled: true,
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "m", text: "Minute" },
  { value: "h", text: "Hour" },
  { value: "d", text: "Day" },
  { value: "w", text: "Week" },
  { value: "M", text: "Month" },
  { value: "q", text: "Quarter" },
  { value: "y", text: "Year" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];

export const ExecutionFrequencyDefinitionOptions = [
  { value: "fixed", text: "Define by fixed interval" },
  { value: "cron", text: "Define by cron expression" },
];

export const DefaultSampleDataSize = 50;
