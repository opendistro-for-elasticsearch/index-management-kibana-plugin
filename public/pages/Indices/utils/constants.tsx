/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { EuiHealth, EuiTableFieldDataColumnType } from "@elastic/eui";
import { ManagedCatIndex } from "../../../../server/models/interfaces";
import { SortDirection } from "../../../utils/constants";

export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: "",
  sortField: "name",
  sortDirection: SortDirection.DESC,
};

const HEALTH_TO_COLOR: {
  [health: string]: string;
  green: string;
  yellow: string;
  red: string;
} = {
  green: "success",
  yellow: "warning",
  red: "danger",
};

export const indicesColumns: EuiTableFieldDataColumnType<ManagedCatIndex>[] = [
  {
    field: "index",
    name: "Index",
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: "250px",
    render: (index: string) => <span title={index}>{index}</span>,
  },
  {
    field: "health",
    name: "Health",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
    render: (health: string, item: ManagedCatIndex) => {
      const color = health ? HEALTH_TO_COLOR[health] : "subdued";
      const text = health || item.status;
      return <EuiHealth color={color}>{text}</EuiHealth>;
    },
  },
  {
    field: "managed",
    name: "Managed by Policy",
    sortable: false,
    truncateText: true,
    textOnly: true,
    align: "right",
    width: "150px",
  },
  {
    field: "status",
    name: "Status",
    sortable: true,
    truncateText: true,
    textOnly: true,
    align: "right",
  },
  {
    field: "store.size",
    name: "Total size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "pri.store.size",
    name: "Primaries size",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "docs.count",
    name: "Total documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
    render: (count: string) => <span title={count}>{count}</span>,
  },
  {
    field: "docs.deleted",
    name: "Deleted documents",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
    render: (deleted: string) => <span title={deleted}>{deleted}</span>,
  },
  {
    field: "pri",
    name: "Primaries",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
  {
    field: "rep",
    name: "Replicas",
    sortable: true,
    truncateText: true,
    textOnly: true,
    dataType: "number",
  },
];
