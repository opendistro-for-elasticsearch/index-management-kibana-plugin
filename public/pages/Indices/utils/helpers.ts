/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import queryString from "query-string";
import { DEFAULT_QUERY_PARAMS } from "./constants";
import { IndicesQueryParams } from "../models/interfaces";

export function getURLQueryParams(location: { search: string }): IndicesQueryParams {
  const { from, size, search, sortField, sortDirection } = queryString.parse(location.search);
  return <IndicesQueryParams>{
    // @ts-ignore
    from: isNaN(parseInt(from, 10)) ? DEFAULT_QUERY_PARAMS.from : parseInt(from, 10),
    // @ts-ignore
    size: isNaN(parseInt(size, 10)) ? DEFAULT_QUERY_PARAMS.size : parseInt(size, 10),
    search: typeof search !== "string" ? DEFAULT_QUERY_PARAMS.search : search,
    sortField: typeof sortField !== "string" ? "index" : sortField,
    sortDirection: typeof sortDirection !== "string" ? DEFAULT_QUERY_PARAMS.sortDirection : sortDirection,
  };
}
