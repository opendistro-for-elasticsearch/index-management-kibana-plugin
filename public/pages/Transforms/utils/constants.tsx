/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { SortDirection } from "../../../utils/constants";

// TODO: Consolidate with Rollup
export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: "",
  sortField: "_id",
  sortDirection: SortDirection.DESC,
};

export const EMPTY_TRANSFORM = JSON.stringify({
  transform: {
    description: "",
    groups: [],
    enabled: true,
    aggregations: {},
    data_selection_query: {},
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];
