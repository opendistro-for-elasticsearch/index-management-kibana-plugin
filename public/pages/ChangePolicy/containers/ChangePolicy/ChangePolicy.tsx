/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { Component } from "react";
import { RouteComponentProps } from "react-router-dom";
import { EuiSpacer, EuiTitle, EuiButton, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
import { CoreStart } from "kibana/public";
import { IndexService, ManagedIndexService } from "../../../../services";
import ChangeManagedIndices from "../../components/ChangeManagedIndices";
import NewPolicy from "../../components/NewPolicy";
import { BREADCRUMBS } from "../../../../utils/constants";
import { ManagedIndexItem } from "../../../../../models/interfaces";
import { getErrorMessage } from "../../../../utils/helpers";
import { PolicyOption } from "../../models/interfaces";

interface ChangePolicyProps extends RouteComponentProps {
  managedIndexService: ManagedIndexService;
  indexService: IndexService;
  core: CoreStart;
}

interface ChangePolicyState {
  selectedPolicies: PolicyOption[];
  selectedManagedIndices: { label: string; value?: ManagedIndexItem }[];
  selectedStateFilters: { label: string }[];
  stateSelected: string;
  stateRadioIdSelected: string;
  managedIndicesError: string;
  selectedPoliciesError: string;
  hasSubmitted: boolean;
}

export enum Radio {
  Current = "current",
  State = "state",
}

export default class ChangePolicy extends Component<ChangePolicyProps, ChangePolicyState> {
  state: ChangePolicyState = {
    selectedPolicies: [],
    selectedManagedIndices: [],
    selectedStateFilters: [],
    stateRadioIdSelected: Radio.Current,
    stateSelected: "",
    managedIndicesError: "",
    selectedPoliciesError: "",
    hasSubmitted: false,
  };

  async componentDidMount(): Promise<void> {
    this.props.core.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.MANAGED_INDICES, BREADCRUMBS.CHANGE_POLICY]);
  }

  onChangeSelectedPolicy = (selectedPolicies: PolicyOption[]): void => {
    // reset the selected state and radio whenever we select a new policy
    const selectedPoliciesError = selectedPolicies.length ? "" : "Required";
    this.setState({ selectedPolicies, selectedPoliciesError, stateSelected: "", stateRadioIdSelected: Radio.Current });
  };

  onChangeManagedIndices = (selectedManagedIndices: { label: string; value?: ManagedIndexItem }[]): void => {
    const managedIndicesError = selectedManagedIndices.length ? "" : "Required";
    if (!selectedManagedIndices.length) {
      this.onChangeStateFilters([]);
    }
    this.setState({ selectedManagedIndices, managedIndicesError });
  };

  onChangeStateFilters = (selectedStateFilters: { label: string }[]): void => {
    this.setState({ selectedStateFilters });
  };

  onChangeStateRadio = (optionId: string): void => {
    this.setState({ stateRadioIdSelected: optionId });
  };

  onStateSelectChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
    this.setState({ stateSelected: e.target.value });
  };

  onChangePolicy = async () => {
    try {
      const { managedIndexService } = this.props;
      const { selectedPolicies, selectedManagedIndices, selectedStateFilters, stateRadioIdSelected, stateSelected } = this.state;

      if (!selectedManagedIndices.length || !selectedPolicies.length) return;

      const indices = selectedManagedIndices.map(
        (selectedManagedIndex: { label: string; value?: ManagedIndexItem }) => selectedManagedIndex.label
      );
      const policyId = selectedPolicies[0].label;
      const state = stateRadioIdSelected === Radio.State ? stateSelected : null;
      const include = selectedStateFilters.map((selectedStateFilter: { label: string }) => ({ state: selectedStateFilter.label }));

      const changePolicyResponse = await managedIndexService.changePolicy(indices, policyId, state, include);

      if (changePolicyResponse.ok) {
        const { updatedIndices, failedIndices, failures } = changePolicyResponse.response;
        if (updatedIndices) {
          this.props.core.notifications.toasts.addSuccess(`Changed policy on ${updatedIndices} indices`);
        }
        if (failures) {
          this.props.core.notifications.toasts.addDanger(
            `Failed to change policy on ${failedIndices
              .map((failedIndex) => `[${failedIndex.indexName}, ${failedIndex.reason}]`)
              .join(", ")}`
          );
        }
      } else {
        this.props.core.notifications.toasts.addDanger(changePolicyResponse.error);
      }
    } catch (err) {
      this.props.core.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem changing policy"));
    }
  };

  onSubmit = async () => {
    const { selectedPolicies, selectedManagedIndices } = this.state;

    const managedIndicesError = selectedManagedIndices.length ? "" : "Required";
    const selectedPoliciesError = selectedPolicies.length ? "" : "Required";

    this.setState({ managedIndicesError, selectedPoliciesError, hasSubmitted: true });
    if (selectedManagedIndices.length && selectedPolicies.length) {
      await this.onChangePolicy();
    }
  };

  render() {
    const { indexService, managedIndexService } = this.props;
    const {
      selectedPolicies,
      selectedManagedIndices,
      selectedStateFilters,
      stateRadioIdSelected,
      stateSelected,
      managedIndicesError,
      selectedPoliciesError,
      hasSubmitted,
    } = this.state;

    return (
      <div style={{ padding: "0px 25px" }}>
        <EuiTitle size="l">
          <h1>Change policy</h1>
        </EuiTitle>

        <EuiSpacer />

        <ChangeManagedIndices
          {...this.props}
          managedIndexService={managedIndexService}
          selectedManagedIndices={selectedManagedIndices}
          selectedStateFilters={selectedStateFilters}
          onChangeManagedIndices={this.onChangeManagedIndices}
          onChangeStateFilters={this.onChangeStateFilters}
          managedIndicesError={hasSubmitted ? managedIndicesError : ""}
        />

        <EuiSpacer />

        <NewPolicy
          {...this.props}
          indexService={indexService}
          selectedPolicies={selectedPolicies}
          stateRadioIdSelected={stateRadioIdSelected}
          stateSelected={stateSelected}
          onChangePolicy={this.onChangeSelectedPolicy}
          onChangeStateRadio={this.onChangeStateRadio}
          onStateSelectChange={this.onStateSelectChange}
          selectedPoliciesError={hasSubmitted ? selectedPoliciesError : ""}
        />

        <EuiSpacer />

        <EuiFlexGroup alignItems="center" justifyContent="flexEnd">
          <EuiFlexItem grow={false}>
            <EuiButton fill onClick={this.onSubmit} data-test-subj="changePolicyChangeButton">
              Change
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </div>
    );
  }
}
