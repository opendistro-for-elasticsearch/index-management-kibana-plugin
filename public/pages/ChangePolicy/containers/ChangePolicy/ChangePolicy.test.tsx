/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, wait } from "@testing-library/react";
import { HashRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import { CoreStart } from "kibana/public";
import ChangePolicy from "./ChangePolicy";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { CoreServicesConsumer, CoreServicesContext } from "../../../../components/core_services";

function renderWithRouter(Component: React.ComponentType<any>) {
  return {
    ...render(
      <Router>
        <Switch>
          <Route
            path={ROUTES.CHANGE_POLICY}
            render={(props) => (
              <CoreServicesContext.Provider value={coreServicesMock}>
                <ServicesContext.Provider value={browserServicesMock}>
                  <ModalProvider>
                    <ServicesConsumer>{(services) => services && <ModalRoot services={services} />}</ServicesConsumer>
                    <CoreServicesConsumer>
                      {(core: CoreStart | null) => (
                        <ServicesConsumer>
                          {({ managedIndexService, indexService }: any) => (
                            <Component indexService={indexService} managedIndexService={managedIndexService} {...props} />
                          )}
                        </ServicesConsumer>
                      )}
                    </CoreServicesConsumer>
                  </ModalProvider>
                </ServicesContext.Provider>
              </CoreServicesContext.Provider>
            )}
          />
          <Redirect from="/" to={ROUTES.CHANGE_POLICY} />
        </Switch>
      </Router>
    ),
  };
}

describe("<ChangePolicy /> spec", () => {
  it("renders the component", async () => {
    const { container } = renderWithRouter(ChangePolicy);

    await wait();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("sets breadcrumbs when mounting", async () => {
    renderWithRouter(ChangePolicy);
    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledTimes(1);
    expect(coreServicesMock.chrome.setBreadcrumbs).toHaveBeenCalledWith([
      BREADCRUMBS.INDEX_MANAGEMENT,
      BREADCRUMBS.MANAGED_INDICES,
      BREADCRUMBS.CHANGE_POLICY,
    ]);
  });
});
