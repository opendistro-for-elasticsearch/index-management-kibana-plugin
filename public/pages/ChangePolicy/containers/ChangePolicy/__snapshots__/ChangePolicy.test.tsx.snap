// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePolicy /> spec renders the component 1`] = `
<div
  style="padding: 0px 25px;"
>
  <h1
    class="euiTitle euiTitle--large"
  >
    Change policy
  </h1>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Choose managed indices
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <div>
            <label
              aria-invalid="false"
              class="euiFormLabel"
              for="some_make_id"
            >
              Managed indices
            </label>
          </div>
          <div
            aria-describedby="some_make_id-help"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                    type="button"
                  >
                    EuiIconMock
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            You can use * as wildcards to form index patterns.
          </div>
        </div>
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <div>
            <label
              class="euiFormLabel"
              for="some_make_id"
            >
              State filters
            </label>
          </div>
          <div
            aria-describedby="some_make_id-help"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox euiComboBox-isDisabled"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Choose state filters
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      disabled=""
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                    disabled=""
                    type="button"
                  >
                    EuiIconMock
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            Apply new policy only on managed indices in these states.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Choose new policy
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            When the new policy will take effect depends on the current state of indices and the states of the new policy.
             
            <a
              class="euiLink euiLink--primary"
              href="https://opendistro.github.io/for-elasticsearch-docs/docs/ism/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn more 
              EuiIconMock
            </a>
          </p>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <div>
            <label
              aria-invalid="false"
              class="euiFormLabel"
              for="some_make_id"
            >
              New policy
            </label>
          </div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      data-test-subj="comboBoxSearchInput"
                      id="some_make_id"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                    type="button"
                  >
                    EuiIconMock
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              checked=""
              class="euiRadio__input"
              id="current"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="current"
            >
              Keep indices in their current state after the policy takes effect
            </label>
          </div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              class="euiRadio__input"
              disabled=""
              id="state"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="state"
            >
              Switch indices to the following state after the policy takes effect
            </label>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <select
                aria-label="Start state for new policy"
                class="euiSelect"
                disabled=""
                id="some_make_id"
              />
              <div
                class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  EuiIconMock
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButton euiButton--primary euiButton--fill"
        data-test-subj="changePolicyChangeButton"
        type="button"
      >
        <span
          class="euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Change
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
